src/wallet/db.h

```diff
@@ -68,7 +68,10 @@ class CDBEnv
     typedef std::pair<std::vector<unsigned char>, std::vector<unsigned char> > KeyValPair;
     bool Salvage(const std::string& strFile, bool fAggressive, std::vector<KeyValPair>& vResult);
 
-    bool Open(const fs::path& path);
+    enum OpenResult { OPEN_OK,
+                        ENV_LOCKED,
+                        OPEN_FAIL };
+    OpenResult Open(const fs::path& path);
```

Comment:
How about `bool Open(const fs::path& path, bool* is_locked = 0)`?

Reply:
Could do, but IMO doesn't simplify the code much, if we're returning something why not use return :)

Reply:
Because you only care of `is_locked` when `Open` fails. For me `ENV_LOCKED` is the *reason* `Open` failed.

Also, you can avoid those `!= OPEN_OK` conditions.

But sure, consider my suggestion a nit.

Reply:
Ah that's true, ok :+1:

Reply:
https://github.com/bitcoin/bitcoin/pull/11904/files#r157363498

Seems fine to use either an enum or a bool, whichever you prefer. The real problem is that VerifyEnvironment is the wrong place for the rename/retry code to live. It creates a bizarre API because you can't reliably open an environment without making two different calls and pointlessly closing and reopening. It's also a bad division of code because the verify function has to make assumptions about what the open function is doing internally.

A better way to organize the code would be to have a single `bool CDBEnv::Open(fs::path, bool retry=false)` function, where setting the `retry` parameter to true renames the log directory and reopens if the initial open fails. Verifying would then just call `Open(path, true)` followed by a `Close`. Anyway this would be nice to clean up, but no need to do it here as part of fixing the lock issue.